src/main/java/edu/harvard/iq/dataverse/ingest/IngestUtil.java

```diff
@@ -100,6 +101,80 @@ public static String duplicateFilenameCheck(FileMetadata fileMetadata, Set<Strin
         return fileName;
     }
 
+    /**
+     * Given a new proposed label or directoryLabel for a file, check against
+     * existing files if a duplicate directoryLabel/label combination would be
+     * created.
+     *
+     * @param label The new label (filename) that is being proposed. Can be
+     * null.
+     * @param directoryLabel The new directoryLabel (file path) that is being
+     * proposed. Can be null.
+     * @param fileMetadatas The list fileMetadatas to be compared against,
+     * probably from a draft.
+     * @param dataFile The file that is being updated with a new name or path.
+     * @return true if there is a conflict, false otherwise.
+     */
+    public static boolean conflictsWithExistingFilenames(String label, String directoryLabel, List<FileMetadata> fileMetadatas, DataFile dataFile) {
+        List<String> filePathsAndNames = new ArrayList<>();
+        for (FileMetadata fileMetadata : fileMetadatas) {
```

Comment:
would it be possible to combine this with findDuplicateFilenames so that the UI and api are using the same method? You're passing in the filemetadata list from the dataset version which is the param for findDuplicateFilenames

Reply:
@sekmiller good idea. Fixed in 8d6a5c8e6.